### PR TITLE
ROS1 -> ROS2 migration doc: Sections on services etc.

### PR DESCRIPTION
**Pull request by nnmm**

### CODE REVIEW

File: docs/porting-to-ROS2.md [hunk 1]
```diff
@@ -147,7 +147,33 @@ There is a [migration guide](https://index.ros.org/doc/ros2/Tutorials/Launch-fil
...
+So you're left with using a [callback](http://docs.ros2.org/foxy/api/rclcpp/classrclcpp_1_1Client.html#a62e48edd618bcb73538bfdc3ee3d5e63). Unfortunately that leaves you with no option to handle failure: For instance, if the service dies, your callback will never get called. There's no easy way to say that you only want to wait for 2 seconds for a result.
+
+Another idea for a workaround is to do something similar to what is done in the `rclcpp::spin_until_future_complete()` function by ourselves. Another possible avenue is using multithreaded executors, see [this post](https://answers.ros.org/question/343279/ros2-how-to-implement-a-sync-service-client-in-a-node/) for some more detail.
+
+
+### Logging
+The node name is now automatically prepended to the log message, so that part can be removed.
+
+
+### `dynamic_reconfigure`
```
Comment (fred-apex-ai) [inline docs/porting-to-ROS2.md]: There is more to be said on porting this, I will document that in a separate PR once I have it figured out better. Still learning about it as we speak

Comment (fred-apex-ai) [reply docs/porting-to-ROS2.md]: And @TakaHoribe wrote a lot more about dynamic_reconfigure but with a different focus in https://github.com/tier4/AutowareArchitectureProposal/pull/78. We should agree on what's needed in the standup and consolidate

**Comment by nnmm** [comment]
ping @esteve @fred-apex-ai – I have removed the section on dynamic_reconfigure since https://github.com/tier4/AutowareArchitectureProposal/pull/78 deals with that in more detail

**Review by fred-apex-ai — APPROVED** [review]
@nnmm I would say this is the smaller section and the PR was opened first. So go ahead, then @TakaHoribe hopefully will not have too much trouble to merge these changes with his

File: docs/porting-to-ROS2.md (hunk 1 above)
Comment (nnmm) [reply docs/porting-to-ROS2.md]: That section is already removed because Taka has a more extensive note on that topic.

**Review by mitsudome-r — APPROVED** [review]
LGTM

**Pull request merged**